tutorials/llm/slimpajama/pretraining.ipynb

```diff
@@ -0,0 +1,171 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Pretraining using Slimpajama\n",
+    "\n",
+    "Let's see how we can use the data generated from the [data pipeline notebook](./data_pipeline.ipynb) to pretrain a model. All we need to do is define the data module based on the generated data and replace it with the mock data module provided by default in the [NeMo llm recipes](../../../nemo/collections/llm/recipes/__init__.py)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import nemo_run as run\n",
+    "from typing import Optional\n",
+    "import pytorch_lightning as pl\n",
+    "from nemo.collections import llm\n",
+    "from nemo.collections.common.tokenizers import SentencePieceTokenizer"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Defining the data module\n",
+    "To define the data module, we can use `llm.PreTrainingDataModule` and pass in the data paths and tokenizer. In case you don't have either of the two, please refer to [data pipeline notebook](./data_pipeline.ipynb). You can also look at the definition of the data module for the other parameters supported like `split`, `num_workers`, `index_mapping_dir`, etc."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def slimpajama(\n",
+    "    gbs: int = 256,\n",
+    "    mbs: int = 4,\n",
+    "    seq_length: int = 8192,\n",
+    ") -> run.Config[pl.LightningDataModule]:\n",
+    "\n",
+    "    return run.Config(\n",
+    "        llm.PreTrainingDataModule,\n",
+    "        paths=[\"/data/slimpajama_megatron/concatenated_chunk1.jsonl_text_document\"],\n",
+    "        seq_length=seq_length,\n",
+    "        global_batch_size=gbs,\n",
+    "        micro_batch_size=mbs,\n",
+    "        tokenizer=run.Config(SentencePieceTokenizer, model_path=\"/data/tokenizer/tokenizer.model\"),\n",
+    "        split=\"99,8,2\",\n",
+    "        num_workers=2,\n",
+    "        index_mapping_dir=\"/data/index_mapping\",\n",
+    "    )"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Configuring the recipe and launching pretraining\n",
```

Comment:
change procedural title to an imperative verb (no "ing" gerunds) and fix capitalization

Configure the Recipe and Launch Pretraining